controllers/openstackcluster_controller.go

```diff
@@ -740,6 +740,33 @@ func reconcilePreExistingNetworkComponents(scope *scope.WithLogger, networkingSe
 		setClusterNetwork(openStackCluster, network)
 	}
 
+	// add pre-existing router to status if router spec is defined
+	if openStackCluster.Status.Router == nil {
+		openStackCluster.Status.Router = &infrav1.Router{}
+	}
+
+	if openStackCluster.Spec.Router != nil {
+		router, err := networkingService.GetRouterByParam(openStackCluster.Spec.Router)
+		if err != nil {
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to find router: %w", err), false)
+			return fmt.Errorf("error fetching cluster router: %w", err)
+		}
+
+		scope.Logger().V(4).Info("Found pre-existing router", "id", router.ID, "name", router.Name)
+
+		routerIPs := []string{}
+		for _, ip := range router.GatewayInfo.ExternalFixedIPs {
+			routerIPs = append(routerIPs, ip.IPAddress)
+		}
+
+		openStackCluster.Status.Router = &infrav1.Router{
+			Name: router.Name,
+			ID:   router.ID,
+			Tags: router.Tags,
+			IPs:  routerIPs,
+		}
+	}
+
```

Comment:
While we are here, I'll like someone (maybe you via this PR, if you don't mind) to document `reconcileProvisionedNetworkComponents` as well. For someone reading these blocks for the first time, it can be confusing why we have these 2 similar functions. Thanks

Reply:
I added a comment that is similar to the comment for `reconcilePreExistingNetworkComponents`.

So we have the same discussion about plural s or not here, but I will adjust it to match the one on `reconcilePreExistingNetworkComponents` after we discussed how this should look like.